plugins/inputs/tail/tail_test.go

```diff
@@ -1030,3 +1060,198 @@ func TestInitInitialReadOffset(t *testing.T) {
 		})
 	}
 }
+
+// TestTailNoLeak tests that we don't leak file descriptors when repeatedly
+// tailing the same file across multiple Gather calls
+func TestTailNoLeak(t *testing.T) {
+	// Create a temp directory for our test file
+	tempDir := t.TempDir()
+	logFile := filepath.Join(tempDir, "test.log")
+
+	content := "cpu usage_idle=100\r\n"
+	require.NoError(t, os.WriteFile(logFile, []byte(content), 0600))
+
+	// Setup the plugin
+	tt := newTestTail()
+	tt.Log = testutil.Logger{}
+	tt.InitialReadOffset = "beginning"
+	tt.Files = []string{logFile}
+	tt.SetParserFunc(newInfluxParser)
+	require.NoError(t, tt.Init())
+
+	// Start the plugin
+	var acc testutil.Accumulator
+	require.NoError(t, tt.Start(&acc))
+	defer tt.Stop()
+
+	// Wait for the plugin to process the file using Gather call
+	require.NoError(t, acc.GatherError(tt.Gather))
+
+	// Wait for the initial metrics
+	require.Eventually(t, func() bool {
+		return acc.NMetrics() >= 1
+	}, time.Second, 100*time.Millisecond, "Did not receive initial metric")
+
+	// Make sure we got the first metric
+	acc.AssertContainsFields(t, "cpu",
+		map[string]interface{}{
+			"usage_idle": float64(100),
+		})
+
+	// Call Gather multiple times to simulate multiple collection intervals
+	require.NoError(t, acc.GatherError(tt.Gather))
+	require.NoError(t, acc.GatherError(tt.Gather))
+	require.NoError(t, acc.GatherError(tt.Gather))
+
+	// Verify we only have one tailer for the file, not multiple
+	// Use the mutex to safely check the tailer count
+	tt.tailersMutex.RLock()
+	tailerCount := len(tt.tailers)
+	tt.tailersMutex.RUnlock()
+	require.Equal(t, 1, tailerCount, "Expected only one tailer despite multiple Gather calls")
+
+	// Reset the accumulator to make it easier to test for the new value
+	acc.ClearMetrics()
+
+	// Stop the current tailer to avoid race conditions
+	tt.Stop()
```

Comment:
Hmmm this might influence the testing result... Imagine in a future PR we accidentally are not closing the tailers correctly during normal processing but _do_ close them on `Stop()`. Your test would still work pass but the code is broken.
You don't need to worry about getting partial data as the underlying `tail` library will only return whole lines (i.e. it blocks until it sees a newline).

Reply:
Thanks for catching that! You're right, by calling tt.Stop() between test phases, I was potentially masking bugs that could occur during normal operation. I've updated both tests accordingly.